src/Deployment.js

```diff
@@ -15,41 +16,54 @@ function Deployment(project, options) {
   this.stderr = '';
   this.success = null;
   this.project = project;
-  this.options = options || {};
+  this.projectConfig = config.PROJECTS[project];
+  this.options = options;
 }
 util.inherits(Deployment, EventEmitter);
 
-Deployment.prototype.python = path.resolve('venv', 'bin', 'python');
-Deployment.prototype.deployScript = path.resolve('deploy.py');
-
-Deployment.prototype.run = function() {
-  var proc = childProcess.spawn(this.python, [this.deployScript, this.project]);
-  proc.stdout.setEncoding('utf8');
-  proc.stderr.setEncoding('utf8');
-
-  proc.stdout.on('data', function(data) {
+Deployment.prototype.run = function run() {
+  var onStdout = function onStdout(data) {
     this.stdout += data;
     this.emit('stdout', data);
-  }.bind(this));
+  }.bind(this);
 
-  proc.stderr.on('data', function(data) {
+  var onStderr = function onStderr(data) {
     this.stderr += data;
     this.emit('stderr', data);
-  }.bind(this));
-
-  proc.on('close', function(code) {
-    this.success = code === 0;
+  }.bind(this);
 
-    if (this.success) {
+  var tasks = [];
+  this.projectConfig.tasks.deploy.forEach(function(task) {
+    tasks.push('cd ' + this.projectConfig.path + ' && ' + task);
+  }.bind(this));
+  return fs
+    .readFileAsync(config.PATH_TO_PRIVATE_KEY)
+    .bind(this)
+    .then(function(key) {
+      return ssh.connect({
+        host: this.projectConfig.hostname,
+        username: this.projectConfig.user,
+        onStdout: onStdout,
+        onStderr: onStderr,
+        debug: true,
+        privateKey: key
+      });
+    })
+    .then(function(connection) {
+      return connection.exec(tasks);
+    })
+    .spread(function(code, stdout, stderr) {
+      this.success = code === 0;
```

Comment:
Maybe, I have to do some changes to promised-ssh. I'll remove it in another PR later on.